noxfile.py

```diff
@@ -116,18 +118,13 @@ def run_scripts(session):
 def set_dev(session):
     """Install PyBaMM in editable mode."""
     set_environment_variables(PYBAMM_ENV, session=session)
-    envbindir = session.bin
-    session.install("-e", ".[all]", silent=False)
-    session.install("cmake", silent=False)
+    session.install("virtualenv", "cmake")
+    session.run("virtualenv", os.fsdecode(VENV_DIR), silent=True)
+    python = os.fsdecode(VENV_DIR.joinpath("bin/python"))
+    session.run(python, "-m", "pip", "install", "-e", ".[all,dev]", silent=False)
```

Comment:
```suggestion
    session.run(python, "-m", "pip", "install", "-e", ".[all,dev]", external=True)
```

We don't require `silent=False` for `session.run()` because it is the default behaviour, could you add `external=True` instead?